### DISCUSSION TITLE
Intellisense support for Zed

### DISCUSSION THREAD

**Discussion by nyomansunima** [tailwindlabs/tailwindcss#14488] [Ideas]
Hi,

I'd like to suggest a thing that saves and helps other people when developing something using Zed code editor. From my experience when using Visual Studio Code is easier to work on the project when the auto-complete comes when forget the utility class.

Now when moving into Zed, I need to remember the utility class and sometimes check the documentation. It's not effective to work as a software engineer.

It would be much appreciated if there any extension came into Zed later to support the workflow 🥳 

_Related_
- [Zed](https://zed.dev/)

Thanks, you're awesome 💖

**Comment by thecrypticace** [comment]
Hey! Our intellisense already works in Zed — they have builtin support for the Tailwind CSS language server. There isn't an _intellisense_ extension that we manage so not all the same features are there (like sorting class selections) but autocomplete and hovers and diagnostics should definitely all work.

(I actually use Zed as my main editor so hopefully I'd notice if something was busted)

If you have a project that isn't working would you mind opening a reproduction here? https://github.com/tailwindlabs/tailwindcss-intellisense

**Reply by laneme** [reply]
Sometimes its hard to reproduce. But i can tell you the reproduction rate goes up tenfold if you use remote projects, but its possible that is caused by a different problem.

I was working the other day on a turbo starter with little code, made a change to tailwind config (_i still use config file for v4_) and tailwind autocomplete stopped working for good. Reverting back all changes, cleaning project (node_modules,turbo etc), language server/editor restart didnt work at all. Hovering over classnames does show class details though.

It happens frequently, especially when on remote projects.

**Reply by thecrypticace** [reply]
Oh interesting. If you can provide a repro project I can test with (doesn't need to be minimal) and some details about the setup you're using I would be super grateful.

**Reply by laneme** [reply]
Is in a rush so wiped my servers but i will try to help with repros when i get the time
